pages/index.tsx

```diff
@@ -1,16 +1,31 @@
 import type { NextPage } from 'next'
 import { Box, Container, Heading } from '@chakra-ui/react'
+import { useAccount, useConnect, useEnsName } from 'wagmi'
+import { InjectedConnector } from 'wagmi/connectors/injected'
 
+function Profile() {
+  const { address, isConnected } = useAccount()
+  const { data: ensName } = useEnsName({ address })
+  const { connect } = useConnect({
+    connector: new InjectedConnector(),
+  })
+
+  if (isConnected) return <div>Connected to {ensName ?? address}</div>
```

Comment:
This might give you a hydration error.

Please implement a `useMounted` hook and only return the address when it is returning `true`.

Example: `if (isMounted && isConnected)` instead of just checking for `isConnected`.

- [`useMounted` hook ready to copy and paste](https://github.com/henkaku-center/nft-karuta-frontend/blob/main/hooks/useMounted.ts)
- [Import the hook and use it in your Profile component](https://github.com/henkaku-center/nft-karuta-frontend/blob/main/pages/index.tsx)

The Profile component should probably be on its own file. But I think that can be a follow-up PR.